File: DevExpress VCL/ExpressQuantumGrid 6/Demos/Delphi/ViewNestedBandsDemo/ViewNestedBandsDemoData.pas
unit ViewNestedBandsDemoData;

interface

uses
  SysUtils, Classes, DB, cxStyles, ImgList, Controls,
  cxGridTableView, cxGridBandedTableView, cxClasses,Forms, DBTables,
  DemoUtils;

type
  TViewNestedBandsDemoDataDM = class(TDataModule)
    dsCars: TDataSource;
    dsOrders: TDataSource;
    dsCustomers: TDataSource;
    DataBase: TDataBase;
    tblCars: TTable;
    tblOrders: TTable;
    tblCustomers: TTable;
    tblCustomersID: TIntegerField;
    tblCustomersFirstName: TStringField;
    tblCustomersLastName: TStringField;
    tblCustomersCompany: TStringField;
    tblCustomersPrefix: TStringField;
    tblCustomersTitle: TStringField;
    tblCustomersAddress: TStringField;
    tblCustomersCity: TStringField;
    tblCustomersState: TStringField;
    tblCustomersZipCode: TStringField;
    tblCustomersSource: TStringField;
    tblCustomersCustomer: TStringField;
    tblCustomersHomePhone: TStringField;
    tblCustomersFaxPhone: TStringField;
    tblCustomersSpouse: TStringField;
    tblCustomersOccupation: TStringField;
    tblCustomersDescription: TMemoField;
    StyleRepository: TcxStyleRepository;
    PaymentTypeImages: TImageList;
    tblOrdersID: TAutoIncField;
    tblOrdersCustomerID: TIntegerField;
    tblOrdersProductID: TIntegerField;
    tblOrdersPurchaseDate: TDateTimeField;
    tblOrdersTime: TDateTimeField;
    tblOrdersPaymentType: TStringField;
    tblOrdersDescription: TMemoField;
    tblOrdersQuantity: TIntegerField;
    tblOrdersPaymentAmount: TCurrencyField;
    tblCarsID: TAutoIncField;
    tblCarsTrademark: TStringField;
    tblCarsModel: TStringField;
    tblCarsHP: TSmallintField;
    tblCarsLiter: TFloatField;
    tblCarsCyl: TSmallintField;
    tblCarsTransmissSpeedCount: TSmallintField;
    tblCarsTransmissAutomatic: TStringField;
    tblCarsMPG_City: TSmallintField;
    tblCarsMPG_Highway: TSmallintField;
    tblCarsCategory: TStringField;
    tblCarsDescription: TMemoField;
    tblCarsHyperlink: TStringField;
    tblCarsPicture: TBlobField;
    tblCarsPrice: TFloatField;
    tblCarsCar: TStringField;
    tblCustomersName: TStringField;
    cxStyle31: TcxStyle;
    cxStyle35: TcxStyle;
    procedure tblCarsCalcFields(DataSet: TDataSet);
    procedure tblCustomersCalcFields(DataSet: TDataSet);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  ViewNestedBandsDemoDataDM: TViewNestedBandsDemoDataDM;

implementation

{$R *.dfm}

procedure TViewNestedBandsDemoDataDM.tblCarsCalcFields(DataSet: TDataSet);
begin
  SetStringFieldValue(tblCarsCar,
    tblCarsTrademark.Value + ' ' + tblCarsModel.Value);
end;

procedure TViewNestedBandsDemoDataDM.tblCustomersCalcFields(
  DataSet: TDataSet);
begin
  SetStringFieldValue(tblCustomersName,
    tblCustomersFirstName.Value + ' ' + tblCustomersLastName.Value);
end;

end.

File: DevExpress VCL/ExpressQuantumGrid 6/Demos/CBuilder/ViewNestedBandsDemo/ViewNestedBandsDemoData.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "ViewNestedBandsDemoData.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "cxClasses"
#pragma link "cxGridTableView"
#pragma link "cxStyles"
#pragma link "cxDBEditRepository"
#pragma link "cxEdit"
#pragma link "cxEditRepositoryItems"
#pragma link "cxGridCardView"
#pragma resource "*.dfm"
TViewNestedBandsDemoDataDM *ViewNestedBandsDemoDataDM;
//---------------------------------------------------------------------------
__fastcall TViewNestedBandsDemoDataDM::TViewNestedBandsDemoDataDM(TComponent* Owner)
  : TDataModule(Owner)
{
}
//---------------------------------------------------------------------------

void __fastcall TViewNestedBandsDemoDataDM::tblCarsCalcFields(TDataSet *DataSet)
{
  tblCarsCar->AsString = tblCarsTrademark->AsString + " " +
    tblCarsModel->AsString;
}
//---------------------------------------------------------------------------

void __fastcall TViewNestedBandsDemoDataDM::tblCustomersCalcFields(TDataSet *DataSet)
{
  tblCustomersName->AsString = tblCustomersFirstName->AsString + " " +
    tblCustomersLastName->AsString;
}
//---------------------------------------------------------------------------

